### PR TITLE
build, net: disable implicit fallthough

### PR DESCRIPTION
**Pull request by tchaikov**
this series disables implicit fallthough and add [[fallthough]] when appropriate

### CODE REVIEW

**Comment by tchaikov** [comment]
in the same spirit of https://github.com/scylladb/scylladb/pull/14607

**Comment by tchaikov** [comment]
this changeset includes #1735 to address the build failure. despite that #1735 is not relevant to this changeset.

File: cmake/CheckHeaders.cmake [hunk 1]
```diff
@@ -84,6 +84,11 @@ function (seastar_check_self_contained library)
   list (REMOVE_DUPLICATES includes)
   target_include_directories (${check_lib}
     PRIVATE ${includes})
+  get_target_property (system_includes ${library} INTERFACE_SYSTEM_INCLUDE_DIRECTORIES)
+  if (system_includes)
+    target_include_directories (${check_lib} SYSTEM
+      PRIVATE ${system_includes})
```
Comment (avikivity) [inline cmake/CheckHeaders.cmake:90]: Maybe we can disable those warnings with #pragma instead, to avoid silencing all warnings.

Comment (tchaikov) [reply cmake/CheckHeaders.cmake]: yeah, even better.

**Comment by tchaikov** [comment]
v2

- use `#pragma` instead of using `-isystem <dir>` compiler command line option to silence warnings

@avikivity changed as suggested. could you take another look?

**Comment by tchaikov** [comment]
@avikivity ping again?

**Comment by tchaikov** [comment]
@avikivity hi Avi, yet another ping?

**Pull request closed without merging**